pkg/scalers/honeycomb_scaler.go

```diff
@@ -0,0 +1,300 @@
+	Query               map[string]interface{} `keda:"name=query, order=triggerMetadata, optional"`
+	QueryRaw            string                 `keda:"name=queryRaw, order=triggerMetadata, optional"`
+	ResultField         string                 `keda:"name=resultField, order=triggerMetadata, optional"`
+	ActivationThreshold float64                `keda:"name=activationThreshold, order=triggerMetadata, default=0"`
+	Threshold           float64                `keda:"name=threshold, order=triggerMetadata"`
+	Breakdowns          []string               `keda:"name=breakdowns, order=triggerMetadata, optional"`
+	Calculation         string                 `keda:"name=calculation, order=triggerMetadata, default=COUNT"`
+	Limit               int                    `keda:"name=limit, order=triggerMetadata, default=1"`
+	TimeRange           int                    `keda:"name=timeRange, order=triggerMetadata, default=60"`
+	TriggerIndex        int
+}
+
+func NewHoneycombScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	logger := InitializeLogger(config, fmt.Sprintf("%s_scaler", honeycombScalerName))
+
+	meta, err := parseHoneycombMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing honeycomb metadata: %w", err)
+	}
+
+	logger.Info("Initializing Honeycomb Scaler", "dataset", meta.Dataset)
+
+	return &honeycombScaler{
+		metricType: metricType,
+		metadata:   meta,
+		// Query Results cannot take longer than 10 seconds to run, see https://api-docs.honeycomb.io/api for details
+		httpClient: &http.Client{Timeout: 15 * time.Second},
+		logger:     logger,
+	}, nil
+}
+
+func parseHoneycombMetadata(config *scalersconfig.ScalerConfig) (honeycombMetadata, error) {
+	meta := honeycombMetadata{}
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, fmt.Errorf("error parsing honeycomb metadata: %w", err)
+	}
+	meta.TriggerIndex = config.TriggerIndex
+
+	// Use queryRaw if provided, else build query from legacy fields
+	if raw, ok := config.TriggerMetadata["queryRaw"]; ok && raw != "" {
+		var q map[string]interface{}
+		if err := json.Unmarshal([]byte(raw), &q); err != nil {
+			return meta, fmt.Errorf("error parsing queryRaw: %w", err)
+		}
+		meta.Query = q
+	} else if meta.Query == nil {
+		q := make(map[string]interface{})
+		if len(meta.Breakdowns) > 0 {
+			q["breakdowns"] = meta.Breakdowns
+		}
+		if meta.Calculation != "" {
+			q["calculations"] = []map[string]string{{"op": meta.Calculation}}
+		}
+		if meta.Limit > 0 {
+			q["limit"] = meta.Limit
+		}
+		if meta.TimeRange > 0 {
+			q["time_range"] = meta.TimeRange
+		}
+		meta.Query = q
+	}
+	if meta.Query == nil {
+		return meta, errors.New("no valid query provided in 'queryRaw', 'query', or legacy fields")
+	}
+	return meta, nil
+}
+
+func (s *honeycombScaler) Close(context.Context) error { return nil }
```

Comment:
We should close idle connections on closing
```suggestion
func (s *honeycombScaler) Close(context.Context) error {
	if s.httpClient != nil {
		s.httpClient.CloseIdleConnections()
	}
	return nil
}
```